### ISSUE TITLE
CODEOWNERS file

### ISSUE THREAD

**Issue by ARMmaster17** [CS4850Group5A/VirtuDoc#89]
TL;DR: Assign people to different parts of the repository, and PRs affecting their part will require their approval before merging in contrast to "anyone can approve".

This is less of a formal work ticket, and more of a proposal of a change to how we handle pull requests in this repository. Up till now anyone can approve any pull request. This has led to a few situations (which are completely my fault and I take full responsibility for those mistakes) where a lot of people are left unaware of what is happening in the repository when PRs get opened and merged very quickly. Thus, my proposal to introduce the [CODEOWNERS file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) to this repo.

## What is it?
The `./.github/CODEOWNERS` file automatically requests PR reviews from people that are labeled as owners of a particular file/folder/extension type if that PR changes one or more of those files. For example, if the codeowners file has the line
```
worker-stuff/*.txt @exampleuser8342834883
```
Then any time a PR is submitted that modifies a TXT file in the `worker-stuff/` directory, `exampleuser8342834883` will be marked as a required PR reviewer that must give an approving review before the PR can be merged. Generally speaking, it's typically silly though to have one person as the owner of something in a repository. GitHub also allows marking a team as owners of something. For example, you can mark the VirtuDoc/Frontend team as responsible for any file ending in `*.js`. When this is the case, anyone on the frontend team can give an approving PR review to merge the build.

## Why do we need it?

Below are my reasons why I think this will help us work better together:
1. People will be kept in the loop about changes to files that affect them (e.g. frontend people are aware of changes to global stylesheet changes for example).
2. PRs affecting multiple parts of the application can be reviewed by multiple people, each having expertise/knowledge over the parts that affect them (and instead of just increasing the number of required reviewers, this change will ensure that small PRs only affecting one part of the application only need one review to merge).
3. Auto-merging of PRs from Dependabot would be really nice, but I don't feel comfortable with Docker manifest changes getting approved without a Docker person looking at it as those changes can't be fully tested in development or in the GHA pipeline, and require a fully manual human review in an MVP sandbox environment.

## Implementation

- [ ] Add CODEOWNERS file to `.github/CODEOWNERS`
- [ ] (Optional) Create teams within VirtuDoc organization on GitHub.
- [ ] Meet with each team to discuss ownership/responsibility over different parts of repository.

## Action items

Looking for thoughts and opinions on this idea before implementing as it does impact our workflow.
